keras/integration_test/distributed_training_test.py

```diff
@@ -73,22 +76,41 @@ def dataset_fn(input_context):
         with strategy.scope():
             model = tf.keras.Sequential([tf.keras.layers.Dense(10)])
             optimizer = tf.keras.optimizers.SGD()
-            model.compile(optimizer, loss="mse", steps_per_execution=10)
+            model.compile(optimizer, loss="mse", steps_per_execution=5)
 
         x = tf.keras.utils.experimental.DatasetCreator(dataset_fn)
 
+        logdir = os.path.join(self.get_temp_dir(), "logdir")
         model.fit(
             x,
             epochs=2,
-            steps_per_epoch=10,
+            steps_per_epoch=20,
             callbacks=[
                 tf.keras.callbacks.TensorBoard(
+                    logdir,
                     update_freq=5,
                     write_steps_per_second=True,
                 )
             ],
         )
 
+        events = []
+        for event_file in glob.glob(logdir + "/train/events.out.*"):
+            for event in tf.compat.v1.train.summary_iterator(event_file):
+                if not event.summary:
+                    continue
+                for value in event.summary.value:
+                    if value.tag != "batch_loss":
+                        continue
+                    events += [event.step]
+        events.sort()
+
+        if not isinstance(
+            strategy, tf.distribute.experimental.ParameterServerStrategy
+        ):
+            # total steps = epochs * steps_per_epoch
+            self.assertEqual(events, [5, 10, 15, 20, 25, 30, 35, 40])
```

Comment:
This is currently failing for `MultiWorkerMirrored2x1CPU`, what's the best way to run this test only for the chief and not the workers?

Reply:
fyi, the way to tell whether it's chief is described [here](https://www.tensorflow.org/tutorials/distribute/multi_worker_with_keras#model_saving_and_loading) (look for `def _is_chief`)

Reply:
The assertion appears to be running on both the worker and the chief, it passes for the chief but fails for the worker as it generates no logs.

This could also be a problem with temp directories, `self.get_temp_dir()` is returning a different location for the chief and the worker. How did other tests deal with testing OS level functionality for distributed strategies?

Reply:
Ah, I see, for assertion of summary file it indeed is fine to check whether it's chief and only assert if it's chief.

For a test like this that's using `tf.__internal__.test.combinations.combine`, there is not an easy way to inject a consistent directory for chief/non-chief. For a test that's using `MultiProcessRunner`, such as [this](https://github.com/keras-team/keras/blob/a8467f50815a1773be1a14ce89716c3d12c500f6/keras/integration_test/mwms_multi_process_runner_test.py#L66), it's possible to provide it as an argument to the chief and workers.

So in this test, let's just assert if it's chief and not worry much. Thanks!